strawberry/schema/types/object_type.py

```diff
@@ -28,10 +28,17 @@ def _get_object_type_for_type_definition(
                 _get_object_type_for_type_definition(interface, type_map)
                 for interface in type_definition.interfaces
             ]
-            # this tells GraphQL core what the returned object's actual type is
-            kwargs["is_type_of"] = lambda obj, _: isinstance(  # type: ignore
-                obj, type_definition.origin
-            )
+
+            # After https://github.com/graphql/graphql-spec/pull/373/files
+            # Now interfaces can also implement interfaces.
+            # But `GraphQLInterfaceType` constructor doesn't accept
+            # `is_type_of` argument.
+            # So we check first, and only add it for `GraphQLInterfaceType`
+            if TypeClass == GraphQLObjectType:
```

Comment:
this looks good, but I'd change the check to be `if type_definition.is_interface` :)

Reply:
You mean something like `if not type_definition.is_interface`?

Reply:
yes, sorry 😊

Reply:
Np, just wanted to check. I'll push the changes